explorer/src/components/SolanabeachComponents/LeaderWorldMap.js

```diff
@@ -0,0 +1,454 @@
+        );
+      case "DE1bawNcRJB9rVm3buyMVfr8mBEoyyu73NBovf2oXJsJ":
+        return (
+          <div className="font-weight-500 coralred">Solana Boot Node 4</div>
+        );
+      case "5D1fNXzvv5NjV1ysLjirC4WY92RNsVH18vjmcszZd8on":
+        return <div className="font-weight-500 coralred">Solana Boot Node</div>;
+      default:
+        return (
+          <div className="font-weight-500 coralred">
+            <HashFormat hash={validator.nodePubkey} />
+          </div>
+        );
+    }
+  }
+
+  if (validator.moniker) {
+    return (
+      <a href={validator.website} className="font-weight-500 " title="...">
+        {validator.moniker}
+      </a>
+    );
+  }
+  switch (validator.nodePubkey) {
+    case "CakcnaRDHka2gXyfbEd2d3xsvkJkqsLw2akB3zsN1D2S":
+      return <div className="font-weight-500 ">Solana Boot Node 1</div>;
+    case "7Np41oeYqPefeNQEHSv1UDhYrehxin3NStELsSKCT4K2":
+      return <div className="font-weight-500 ">Solana Boot Node 2</div>;
+    case "GdnSyH3YtwcxFvQrVVJMm1JhTS4QVX7MFsX56uJLUfiZ":
+      return <div className="font-weight-500 ">Solana Boot Node 3</div>;
+    case "DE1bawNcRJB9rVm3buyMVfr8mBEoyyu73NBovf2oXJsJ":
+      return <div className="font-weight-500 ">Solana Boot Node 4</div>;
+    case "5D1fNXzvv5NjV1ysLjirC4WY92RNsVH18vjmcszZd8on":
+      return <div className="font-weight-500 ">Solana Boot Node</div>;
+    default:
+      return (
+        <div className="font-weight-500 ">
+          <HashFormat hash={validator.nodePubkey} />
+        </div>
+      );
+  }
+}
+
+function calcMarkerColor(nodeCount) {
+  if (nodeCount === 1) return "palegreen";
+  if (nodeCount < 5) return "yellowgreen";
+  if (nodeCount < 10) return "limegreen";
+  return "lime";
+}
+
+function calcMarkerRadius(nodeCount) {
+  if (nodeCount === 1) return 10;
+  if (nodeCount < 5) return 11;
+  if (nodeCount < 10) return 12;
+  return 14;
+}
+
+function renderMapLocationLabel(currentLeader) {
+  if (currentLeader == null) return;
+
+  if (currentLeader.location) {
+    if (currentLeader.location.country && currentLeader.location.city) {
+      return (
+        currentLeader.location.country + ", " + currentLeader.location.city
+      );
+    }
+    return currentLeader.location.country;
+  }
+  return "";
+}
+
+export default class LeaderWorldMap extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      root: 0,
+      mapInitialized: false,
+      currentLeader: { location: { country: "", city: "" } },
+      servedSlots: [],
+      nextLeaders: [],
+      markers: [
+        { coordinates: [100, 100], nodeCount: 0, leader: false, pubkeys: [] },
+      ],
+    };
+  }
+
+  componentWillUnmount() {
+    if (this.socket) {
+      this.socket.disconnect();
+    }
+  }
+
+  componentDidMount() {
+    const { socket } = this.props;
+    socket.on("validatorInfo", (RPCdata) => this.handleRPCdata(RPCdata));
+    socket.on("rootNotification", (data) => this.handleNewRoot(data));
```

Comment:
Could you hook this up to the existing context provider? You can add a type for validator info in `providers/stats/solanaBeach.tsx`